server/container_create_linux.go

```diff
@@ -161,10 +164,30 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrfactory.Cont
 		return nil, err
 	}
 
+	isRuntimePullImage := false
+	if sb.RuntimeHandler() != "" {
+		r, ok := s.config.Runtimes[sb.RuntimeHandler()]
+		if ok {
+			isRuntimePullImage = r.RuntimePullImage
+		}
+	}
+
+	var someNameOfTheImage *references.RegistryImageReference
+	var imageID storage.StorageImageID
+	var someRepoDigest string
+	var imageAnnotations map[string]string
+
 	// Get imageName and imageID that are later requested in container status
-	someNameOfTheImage, imageID, someRepoDigest, imageAnnotations, err := s.getInfoFromImage(userRequestedImage)
-	if err != nil {
-		return nil, err
+	if isRuntimePullImage {
```

Comment:
why are we using the artifact functions here?

Reply:
ah @saschagrunert suggested this. it feels weird to me to use the artifact functions here, why did we go this route?

Reply:
If the image is not there, we need a way to access the remote repo's metadata to get all needed info, without pulling the full image.
In a way or another, I need to access the data without pulling the layers. The containers/image API did not allow me to do that (or I missed it?) so Sascha suggested this alternative, which allows me to get just the manifest and config that I need.

Reply:
yeah I think the approach is good  but maybe naming the functions differently so it's clearer that's what we're doing here?

Reply:
Yeah we can split the existing bits if required. :+1: